src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+                s_addBracesLocalizableTitle,
+                s_addBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+
+        private DiagnosticDescriptor CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.RemoveBracesDiagnosticId,
+                s_removeBracesLocalizableTitle,
+                s_removeBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+        public bool RunInProcess => false;
+
+        public override void Initialize(AnalysisContext context)
+        {
+            var syntaxKindsOfInterest =
+            ImmutableArray.Create(SyntaxKind.IfStatement,
+                SyntaxKind.ElseClause,
+                SyntaxKind.ForStatement,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.ForEachComponentStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.DoStatement,
+                SyntaxKind.UsingStatement,
+                SyntaxKind.LockStatement,
+                SyntaxKind.FixedStatement);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, syntaxKindsOfInterest);
+        }
+
+        public DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        public void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var optionSet = context.Options.GetOptionSet();
+            var useBracesOption = optionSet.GetOption(CSharpCodeStyleOptions.UseBracesWherePossible);
+            var severity = useBracesOption.Notification.Value;
+
+            if (useBracesOption.Value)
+            {
+                var diagnostic = GetAddBracesDiagnostic(context.Node, severity);
+                if (diagnostic != null)
+                {
+                    context.ReportDiagnostic(diagnostic);
+                }
+            }
+            else
+            {
+                var diagnostic = GetRemoveBracesDiagnostic(context.Node, severity);
+                if (diagnostic != null)
+                {
+                    context.ReportDiagnostic(diagnostic);
+                }
+            }
+        }
+
+        private Diagnostic GetAddBracesDiagnostic(SyntaxNode node, DiagnosticSeverity severity)
+        {
+            if (node.IsKind(SyntaxKind.IfStatement))
+            {
+                var ifStatement = (IfStatementSyntax)node;
+                if (AnalyzeIfStatement(ifStatement))
+                {
+                    return Diagnostic.Create(_severityToAddBracesDescriptorMap[severity],
+                        ifStatement.IfKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.IfKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.ElseClause))
```

Comment:
this should be "else if".  otherwise you're running this code even if AnalyzeIfStatement above fell through.